app/src/main/java/com/example/paulsuarez/a27_facebook_lite/FriendDetailActivity.java

```diff
@@ -0,0 +1,41 @@
+package com.example.paulsuarez.a27_facebook_lite;
+
+import android.content.Intent;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+
+public class FriendDetailActivity extends AppCompatActivity {
+
+    private TextView name;
+    private TextView bio;
+    private Button home;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_friend_detail);
+
+        Intent data = getIntent();
+        data.getStringExtra("name");
+        data.getStringExtra("bio");
+
+        name = findViewById(R.id.name);
+        name.setText(data.getStringExtra("name"));
+
+        name = findViewById(R.id.bio);
+        name.setText(data.getStringExtra("bio"));
+
+        home = findViewById(R.id.home);
+        home.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                finish();
+            }
+        });
+    }
+}
+
+// this page was the hardest. getting one activity to update with different content is still so abstract for me.
```

Comment:
One way to think about activities and intents is this:
An android app is a bundle of one or more activities. Each activity can be thought of as a sort of "sub program".
Intents are like commands to execute an activity and extras are arguments. It's not unlike running a command in your terminal/command prompt. You first specify the program you want to execute ("git" for example), which is like the intent. Then you specify the arguments that tell git what to do ("checkout foobranch" for example). Those are like extras.